pkg/controller/direct/compute/refs.go

```diff
@@ -0,0 +1,243 @@
+	computeTargetGrpcProxy, err := resolveResource(ctx, reader, src, ref, schema.GroupVersionKind{
+		Group:   "compute.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "ComputeTargetGrpcProxy",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// targetField: self_link
+	// See compute servicemappings for details
+	return &v1alpha1.ResourceRef{
+		External: computeTargetGrpcProxy.GetSelfLink()}, nil
+}
+
+func ResolveComputeTargetHTTPProxy(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*v1alpha1.ResourceRef, error) {
+	computeTargetHTTPProxy, err := resolveResource(ctx, reader, src, ref, schema.GroupVersionKind{
+		Group:   "compute.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "ComputeTargetHTTPProxy",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// targetField: self_link
+	// See compute servicemappings for details
+	return &v1alpha1.ResourceRef{
+		External: computeTargetHTTPProxy.GetSelfLink()}, nil
+}
+
+func ResolveComputeTargetHTTPSProxy(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*v1alpha1.ResourceRef, error) {
+	computeTargetHTTPSProxy, err := resolveResource(ctx, reader, src, ref, schema.GroupVersionKind{
+		Group:   "compute.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "ComputeTargetHTTPSProxy",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// targetField: self_link
+	// See compute servicemappings for details
+	return &v1alpha1.ResourceRef{
+		External: computeTargetHTTPSProxy.GetSelfLink()}, nil
+}
+
+func ResolveComputeTargetSSLProxy(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*v1alpha1.ResourceRef, error) {
+	computeTargetSSLProxy, err := resolveResource(ctx, reader, src, ref, schema.GroupVersionKind{
+		Group:   "compute.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "ComputeTargetSSLProxy",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// targetField: self_link
+	// See compute servicemappings for details
+	return &v1alpha1.ResourceRef{
+		External: computeTargetSSLProxy.GetSelfLink()}, nil
+}
+
+func ResolveComputeTargetTCPProxy(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*v1alpha1.ResourceRef, error) {
+	computeTargetTCPProxy, err := resolveResource(ctx, reader, src, ref, schema.GroupVersionKind{
+		Group:   "compute.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "ComputeTargetTCPProxy",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// targetField: self_link
+	// See compute servicemappings for details
+	return &v1alpha1.ResourceRef{
+		External: computeTargetTCPProxy.GetSelfLink()}, nil
+}
+
+func ResolveComputeTargetVPNGateway(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*v1alpha1.ResourceRef, error) {
```

Comment:
A problem here is that previously we could have used the type of the ref to call the correct Resolver function (by making it a member function).

Reply:
I changed it back